src/coreclr/jit/instr.cpp

```diff
@@ -1230,7 +1230,8 @@ void CodeGen::inst_RV_RV_TT(
     if (IsEmbBroadcast)
     {
         instOptions = INS_OPTS_EVEX_b;
-        if (HWIntrinsicInfo::IsBitwiseOperation(ins) && (op2->AsHWIntrinsic()->Op(1)->TypeIs(TYP_LONG) || op2->AsHWIntrinsic()->Op(1)->TypeIs(TYP_ULONG)))
+        if (HWIntrinsicInfo::IsBitwiseOperation(ins) &&
+            (op2->AsHWIntrinsic()->Op(1)->TypeIs(TYP_LONG) || op2->AsHWIntrinsic()->Op(1)->TypeIs(TYP_ULONG)))
```

Comment:
You can check both simultaneously with:
```suggestion
        if (HWIntrinsicInfo::IsBitwiseOperation(ins) && varTypeIsLong(op2->AsHWIntrinsic()->Op(1)))
```

Reply:
That being said, would this be better:
```suggestion
        if (HWIntrinsicInfo::IsBitwiseOperation(ins) && varTypeIsLong(op2->AsHWIntrinsic()->GetSimdBaseType()))
```

Since it checks the exact type of broadcast intended, irrespective of the type of the underlying operand.

Reply:
Indeed, I missed the basetype information on broadcast node, thanks for the suggestion, updated.